mat64/io.go

```diff
@@ -81,18 +87,24 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 		panic("mat64: unmarshal into non-zero matrix")
 	}
 
+	var tmp [8]byte
 	buf := bytes.NewReader(data)
-	var rows int64
-	err := binary.Read(buf, binary.LittleEndian, &rows)
+	_, err := buf.Read(tmp[:])
```

Comment:
actually, scratch that.
this whole dance b/w `io.Reader` and the scratch buffer `tmp` is pointless and just a remnant of the before-optimization code.

one can just work on the original `data []byte` slice, managing an extra `n int` index into that slice.